src/parser/jobs/pld/modules/Goring.tsx

```diff
@@ -1,62 +1,300 @@
+	private clip: {[key: number]: number} = {
+		[this.data.statuses.GORING_BLADE.id]: 0,
+		[this.data.statuses.BLADE_OF_VALOR.id]: 0,
+	}
+	private tracker: DotApplicationTracker = {}
+	private trackedClipping: boolean = false;
+
+	override initialise() {
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.GORING_BLADE.id), this.onGoringCast)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.BLADE_OF_VALOR.id), this.onValorCast)
+		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.GORING_BLADE.id), this.onGoringApply)
+		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.BLADE_OF_VALOR.id), this.onValorApply)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private createTargetApplicationList() {
+		return {
+			[this.data.statuses.GORING_BLADE.id]: [],
+			[this.data.statuses.BLADE_OF_VALOR.id]: [],
+		}
+	}
+
+	private pushApplication(targetKey: string, statusId: number, event: Events['statusApply'], clip?: number) {
+		const target = this.tracker[targetKey] = this.tracker[targetKey] || this.createTargetApplicationList()
+		const source = this.lastDotCast
+		target[statusId].applications.push({event, clip, source})
+	}
+
+	private onGoringCast(event: Events['action']) {
+		this.lastDotCast = event.action
+	}
+
+	private onValorCast(event: Events['action']) {
+		this.lastDotCast = event.action
+	}
+
+	private onGoringApply(event: Events['statusApply']) {
+		const statusId = event.status
 
-	override addChecklistRules() {
+		// Make sure we're tracking for this target
+		const applicationKey = event.target
+		const trackerInstance = this.tracker[applicationKey] = this.tracker[applicationKey] || {}
+
+		// If neither dot has been applied yet, initialize data and return
+		if (!trackerInstance[statusId] && !trackerInstance[this.data.statuses.BLADE_OF_VALOR.id]) {
+			trackerInstance[statusId] = {
+				lastApplication: event.timestamp,
+				applications: [],
+			}
+			//save the application for later use in the output
+			this.pushApplication(applicationKey, statusId, event)
+			return
+		}
+
+		// If only the opposite DoT has been used on this target, initialize data and move to clip calculation
+		if (!trackerInstance[statusId] && trackerInstance[this.data.statuses.BLADE_OF_VALOR.id]) {
+			trackerInstance[statusId] = {
+				lastApplication: 0,
+				applications: [],
+			}
+		}
+
+		// If Blade of Valor has been used on this target, get most recent previous DoT application timestamp
+		let lastDotApplication = trackerInstance[statusId].lastApplication
+		if (trackerInstance[this.data.statuses.BLADE_OF_VALOR.id]) {
+			lastDotApplication = Math.max(lastDotApplication, trackerInstance[this.data.statuses.BLADE_OF_VALOR.id].lastApplication)
+		}
+		// Base clip calc
+		let clip = this.STATUS_DURATION[statusId] - (event.timestamp - lastDotApplication)
```

Comment:
Prefer a direct lookup rather than STATUS_DURATION, or otherwise name the var more appropriately as detailed above. Personally I think you can forgo the var entirely.